askama/src/filters/alloc.rs

```diff
@@ -776,6 +816,93 @@ fn flush_title(dest: &mut (impl fmt::Write + ?Sized), s: &str) -> fmt::Result {
     Ok(())
 }
 
+/// A prefix usable for indenting [prettified JSON data](super::json::json_pretty) and
+/// [`|indent`](indent)
+///
+/// ```
+/// # use askama::filters::AsIndent;
+/// assert_eq!(4.as_indent(), "    ");
+/// assert_eq!(" -> ".as_indent(), " -> ");
+/// ```
+pub trait AsIndent {
+    /// Borrow `self` as prefix to use.
+    fn as_indent(&self) -> &str;
+}
+
+impl AsIndent for str {
+    #[inline]
+    fn as_indent(&self) -> &str {
+        self
+    }
+}
+
+#[cfg(feature = "alloc")]
+impl AsIndent for alloc::string::String {
+    #[inline]
+    fn as_indent(&self) -> &str {
+        self
+    }
+}
+
+impl AsIndent for usize {
```

Comment:
Would be nice to support all unsigned, no? And also support signed integers too (with an extra check to ensure it's not negative).

Reply:
Then you could not write a simple `2` to for two spaces, when the trait is implemented for signed and unsigned types.

Reply:
Faire enough!